File: chap4/Globale.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let a = ref 3

let f x =
  print_int (2 * x); print_newline ();
  a := 2 * x

let _ =
  a := 3;
  let n = 4 in
  f (!a + n)

File: chap4/Globale.java
// Ces programmes sont sous licence CeCILL-B V1.
class Globale {

  static int a;

  static void f (int x) { 
    System.out.println (2 * x);
    a = 2 * x;}

  public static void main (String [] args) {
    int n;
    a = 3;
    n = 4;
    f(a + n);}}

